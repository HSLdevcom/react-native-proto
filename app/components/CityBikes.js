/**
 * CityBikes component
 * @flow
 */

import React from 'react';
import CustomWebView from './CustomWebView';

export const CITYBIKE_URL = 'https://www.hsl.fi/citybike';

function CityBikes() {
    return <CustomWebView uri={CITYBIKE_URL} />;
}

export default CityBikes;
