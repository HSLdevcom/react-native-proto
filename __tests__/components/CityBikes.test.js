// https://facebook.github.io/jest/blog/2016/07/27/jest-14.html
import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {fromJS} from 'immutable';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import CityBikes from '../../app/components/CityBikes';

const mockStore = configureStore();
// https://github.com/facebook/react-native/issues/12440#issuecomment-282184173
jest.unmock('ScrollView');
const cookies = fromJS({
    cookie: false,
});

it('renders correctly', () => {
    const tree = renderer.create(
        <Provider
            store={mockStore({
                cookies,
                removeCookie: jest.fn(),
                removeSession: jest.fn(),
                setCookie: jest.fn(),
                setSession: jest.fn(),
            })}
        >
            <CityBikes />
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
