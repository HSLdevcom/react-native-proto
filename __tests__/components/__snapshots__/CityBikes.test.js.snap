// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(0, 122, 201)",
        "borderBottomWidth": 1,
        "borderColor": "rgb(0, 122, 201)",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 50,
      },
      Object {
        "height": 1254,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 8,
        },
        Object {
          "height": 80,
          "left": 335,
          "position": "absolute",
          "width": 80,
          "zIndex": 2,
        },
        null,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={undefined}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript="
                // setTimeout(function() {
                //     var e = document.createElement('h1');
                //     e.innerText = navigator.userAgent;
                //     document.getElementsByTagName('body')[0].append(e);
                // }, 1000);
            "
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={undefined}
      scrollEnabled={true}
      source={
        Object {
          "uri": "https://www.hsl.fi/citybike",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "marginTop": 63,
            },
            Object {
              "marginTop": 63,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;
